plugin-locationcomponent/src/test/java/com/mapbox/maps/plugin/locationcomponent/LocationComponentPluginImplTest.kt

```diff
@@ -0,0 +1,354 @@
+  }
+
+  @Test
+  fun testRemoveOnIndicatorPositionChangedListener() {
+    val positionListener = mockk<OnIndicatorPositionChangedListener>(relaxed = true)
+    locationComponentPlugin.addOnIndicatorPositionChangedListener(positionListener)
+    locationComponentPlugin.removeOnIndicatorPositionChangedListener(positionListener)
+    locationComponentPlugin.indicatorPositionChangedListener.onIndicatorPositionChanged(
+      Point.fromLngLat(
+        0.0,
+        0.0
+      )
+    )
+    verify(exactly = 0) { positionListener.onIndicatorPositionChanged(Point.fromLngLat(0.0, 0.0)) }
+  }
+
+  @Test
+  fun testAddOnIndicatorBearingChangedListener() {
+    val bearingListener = mockk<OnIndicatorBearingChangedListener>(relaxed = true)
+    locationComponentPlugin.addOnIndicatorBearingChangedListener(bearingListener)
+    locationComponentPlugin.indicatorBearingChangedListener.onIndicatorBearingChanged(0.0)
+    verify { bearingListener.onIndicatorBearingChanged(0.0) }
+  }
+
+  @Test
+  fun testRemoveOnIndicatorBearingChangedListener() {
+    val bearingListener = mockk<OnIndicatorBearingChangedListener>(relaxed = true)
+    locationComponentPlugin.addOnIndicatorBearingChangedListener(bearingListener)
+    locationComponentPlugin.removeOnIndicatorBearingChangedListener(bearingListener)
+    locationComponentPlugin.indicatorBearingChangedListener.onIndicatorBearingChanged(0.0)
+    verify(exactly = 0) { bearingListener.onIndicatorBearingChanged(0.0) }
+  }
+
+  @Test
+  fun testSetLocationProvider() {
+    val mockLocationProvider = mockk<LocationProvider>(relaxed = true)
+    preparePluginInitialisationWithEnabled()
+    verify { locationProvider.registerLocationConsumer(any()) }
+    locationComponentPlugin.setLocationProvider(mockLocationProvider)
+    verify { locationProvider.unRegisterLocationConsumer(any()) }
+    verify { mockLocationProvider.registerLocationConsumer(any()) }
+    assertEquals(mockLocationProvider, locationComponentPlugin.getLocationProvider())
+  }
+
+  @Test
+  fun testSetLocationProviderDeactivated() {
+    val mockLocationProvider = mockk<LocationProvider>(relaxed = true)
+    locationComponentPlugin.locationProvider = locationProvider
+    locationComponentPlugin.setLocationProvider(mockLocationProvider)
+    verify { locationProvider.unRegisterLocationConsumer(any()) }
+    verify(exactly = 0) { mockLocationProvider.registerLocationConsumer(any()) }
+    assertEquals(mockLocationProvider, locationComponentPlugin.getLocationProvider())
+  }
+
+  @Test
+  fun testOnStyleChanged() {
+    every { locationPuckManager.isLayerInitialised() } returns false
+    preparePluginInitialisationWithEnabled()
+    verify(exactly = 2) { locationPuckManager.isLayerInitialised() }
+    verify(exactly = 1) { locationPuckManager.initialize(style) }
+    locationComponentPlugin.onStyleChanged(style)
+    verify(exactly = 3) { locationPuckManager.isLayerInitialised() }
+    verify(exactly = 2) { locationPuckManager.initialize(style) }
```

Comment:
So it's expected to initialize `locationPuckManager` twice here?

Reply:
The `fun preparePluginInitialisationWithEnabled` has already initialised the puck manager once, when style changes, we need to initialise the location manager again.